Add doc for transform functions before sinks (#543)
apache/pulsar-site: docs/io-use.md

</Tabs>
````

> ## Run a Pulsar Function before a sink connector

Reviewer: @cbornet I don't quite understand this user task from the heading. 
Can you pls also attach the feature PR in this PR's description?

Author: Done. PTAL

> You have the possibility to run a [Pulsar Function](functions-overview.md) in-memory before a sink connector (see [PIP-193](https://github.com/apache/pulsar/issues/16739)).

Reviewer 2: ```suggestion
You can run a [Pulsar Function](functions-overview.md) in memory before a sink connector. For details, see [[PIP 193: Sink preprocessing Function](https://github.com/apache/pulsar/issues/16739).
```

> This provides better latency, less I/O and disk consumption than going through an intermediate topic.

Reviewer: ```suggestion
Running a [Pulsar Function](functions-overview.md) in memory before a sink connector provides lower latency, less I/O, and disk consumption than going through an intermediate topic.
```
I incorporated the first sentence into this one so it can be removed.

> Running a Pulsar Function in memory before a sink connector provides lower latency, less I/O and disk consumption than going through an intermediate topic.

Reviewer 2: ```suggestion
Running a Pulsar Function in memory before a sink connector provides lower latency, less I/O, and disk consumption than going through an intermediate topic.
```
